doc/goals.rst

```diff
@@ -0,0 +1,141 @@
+************
+Goal of dune
+************
+
+The main goal of the Dune project is to provide the best possible
+build tool for the whole of the OCaml community, from individual
+developpers who work on open source projects during their free time
+all the way to larger companies such as Jane Street. And to the extent
+that is reasonably possible, help provide the same feature for friend
+communities such as Coq and possibly Reason/Bucklescript in the
+future.
+
+We haven't reached this goal yet and Dune still lacks in some areas in
+order to be such a tool, but we are slowly working towards that goal.
+On a practical level, there are a few boxes to check. There are in
+fact a ton details to sort out, but at a high-level a tool that works
+for everyone in the OCaml community should at least:
+
+1. have excelent backward compatibility properties
+2. have a robust and scalable core
+3. remain a no-brainer dependency
+4. have very good support for the OCaml language
+5. be extensible
```

Comment:
I just noticed that none of these items really speak to usability or accessibility. But this seems like an important goal for "a high-level a tool that works for everyone in the OCaml community".

Reply:
Indeed, you are right that this is an important point. I'll add something about it.